### DISCUSSION TITLE
Export only filtered data in a q-table Vue 3

### DISCUSSION THREAD

**Discussion by jandresmedina** [quasarframework/quasar#16234] [General - Components / Directives / etc] [answered]
Hello,
I'm new to vue and still learning. What I would like to do is use the export function of a q table but export only the filtered data instead of everything. I am using the function from the example but I don't know how to work/get only the filtered data.

exportTable () {
      // naive encoding to csv format
      const content = [ this.columns.map(col => wrapCsvValue(col.label)) ].concat(
        this.$refs.table.filteredSortedRows.map(row => this.columns.map(col => wrapCsvValue(
          typeof col.field === 'function'
            ? col.field(row)
            : row[col.field === void 0 ? col.name : col.field],
          col.format
        )).join(','))
      ).join('\r\n')

I did it using this.$refs.table.filteredSortedRows.map(....
but I get the following error: Uncaught TypeError: Cannot read properties of undefined (reading '$refs')

Any ideas ?
thank you

**Comment by metalsadman** [comment]
Show us the rest of the code. Or try optional chaining there, my guess is you didn't assign the ref or your accessing it before the component has mounted.

**Reply by jandresmedina** [reply]
Basically I want to do this:
https://codepen.io/metalsadman/pen/LYNzrEE?editors=1010
using the latest version of quasar framework - vue 3, I have not found any updated example for the version that I mention, it would be of great help if you have any updated example for vue 3 - quasar, thanks for answering

**Comment by metalsadman** [comment, marked as answer]
@AndresIMedina https://codepen.io/metalsadman/pen/GRPpRmE?editors=1010

**Reply by jandresmedina** [reply]
works for me, amazing, I'm sure it will help many people who are going through this inconvenience.

thanks for your time, you're great
